### PR TITLE
WIP: RateLimiter

### PR DESCRIPTION
**Pull request by georgekola**

### CODE REVIEW

**Comment by georgekola** [comment]
@indutny please review

File: stud.cc [hunk 1]
```diff
@@ -1677,6 +1878,72 @@ static void handle_accept(struct ev_loop *loop, ev_io *w, int revents) {
...
+static int get_current_epoch()
+{
+    time_t cur_time = time(NULL);
+    bool flag=false;
+    if(cur_time - lastTS > (2 * TWICE_COUNTER_THRESHOLD)){
+        // clear the entire blacklist 
+        BlackListedIP.clear();
+        flag = true;
+    }
+    int epoch = (cur_time & 48) >> 4;
```
Comment (indutny) [inline stud.cc]: Every 16 seconds?

**Comment by indutny** [comment]
Overall appears to be doing the "right thing". Don't see any obvious problems with this approach.

File: stud.cc (hunk 1 above)
Comment (georgekola) [inline stud.cc]: Yes, window of 64 seconds partitioned into 4 one every 16 seconds

Comment (georgekola) [inline stud.cc]: @indutny

**Comment by georgekola** [comment]
@indutny thanks for reviewing

**Pull request closed without merging**